### ISSUE TITLE
Regard to the range with my interesting clustering to draw a line to embrace it 

### ISSUE THREAD

**Issue by honghh2018** [satijalab/seurat#2980]
Hi satijalab,
     The plot showed on some paper like below:
![image](https://user-images.githubusercontent.com/41668708/81380676-d91fe480-913d-11ea-973b-1420c08c1b2d.png)
     I want to draw the circle like the picture show above, and how should i do for it?
     However, DimPlot(pancreas.integrated,reduction = 'tsne',label = T,cols=cols)+geom_curve() showing error.
     Any advice would be appreciated.

**Comment by andrewwbutler** [comment]
Hi,

It looks like [`geom_ellipse`](https://rdrr.io/cran/ggforce/man/geom_ellipse.html) is probably what you're looking for here.

**Comment by cesco-lemon** [comment]
Hello!
Sorry for resuscitating this :)
I tried to build ellipses on PCA plots created for my dataset after AverageExpression and got an error:

! `stat_ellipse()` requires the following missing aesthetics: x and y

Even though I have added + stat_ellipse(inherit.aes = TRUE). Any idea of what I am doing wrong? The DimPlot comes out fine if I omit stat_ellipse and geom_ellipse() gives same error


quick run-throguh:

library(ggplot2)
library(Seurat)
library(dplyr)
library(Matrix)
library(ggforce)

object <-readRDS(file='~/Desktop/object.rds')
object.averages <- AverageExpression(object, return.seurat=TRUE)
object.averages <- NormalizeData(object.averages)
object.averages <- FindVariableFeatures(object.averages)
top20 <- head(VariableFeatures(object.averages), 20)
all.genes <- rownames(object.averages)
object.averages <- ScaleData(object.averages, features = all.genes)
object.averages <- RunPCA(object.averages, features = VariableFeatures(object = object.averages), npcs=7)

DimPlot(object.averages, reduction = "pca", dims=c(1,2)) + stat_ellipse(inherit.aes = TRUE)
